python/lsst/afw/table/base/base.cc

```diff
@@ -0,0 +1,195 @@
+    declareBaseRecordOverloads<std::uint16_t>(cls, "U");
+    declareBaseRecordOverloads<std::int32_t>(cls, "I");
+    declareBaseRecordOverloads<std::int64_t>(cls, "L");
+    declareBaseRecordOverloads<std::string>(cls, "String");
+    declareBaseRecordOverloads<lsst::afw::geom::Angle>(cls, "Angle");
+    declareBaseRecordArrayOverloads<std::uint16_t>(cls, "ArrayU");
+    declareBaseRecordArrayOverloads<int>(cls, "ArrayI");
+    declareBaseRecordArrayOverloads<float>(cls, "ArrayF");
+    declareBaseRecordArrayOverloads<double>(cls, "ArrayD");
+
+    // These are master getters and setters that can take either strings, Keys, or
+    // FunctorKeys, and dispatch to key.get.
+    auto getter = [](py::object const & self, py::object key) -> py::object {
+        py::object schema = self.attr("schema");
+        if (py::isinstance<py::str>(key) || py::isinstance<py::bytes>(key)) {
+            key = schema.attr("find")(key).attr("key");
+        }
+        return key.attr("get")(self);
+    };
+    auto setter = [](py::object const & self, py::object key, py::object const & value) -> void {
+        py::object schema = self.attr("schema");
+        if (py::isinstance<py::str>(key) || py::isinstance<py::bytes>(key)) {
+            key = schema.attr("find")(key).attr("key");
+        }
+        key.attr("set")(self, value);
+    };
+
+    // The distinction between get/set and operator[] is meaningful in C++, because "record[k] = v"
+    // operates by returning an object that can be assigned to.
+    // But there's no meaningful difference between get/set and __getitem__/__setitem__.
+    cls.def("get", getter);
+    cls.def("__getitem__", getter);
+    cls.def("set", setter);
+    cls.def("__setitem__", setter);
+    return cls;
+}
+
+PyBaseTable declareBaseTable(py::module & mod) {
+    PyBaseTable cls(mod, "BaseTable");
+    utils::python::addSharedPtrEquality<BaseTable>(cls);
+    cls.def_static("make", &BaseTable::make);
+    cls.def("getMetadata", &BaseTable::getMetadata);
+    cls.def("setMetadata", &BaseTable::setMetadata, "metadata"_a);
+    cls.def("popMetadata", &BaseTable::popMetadata);
+    cls.def("makeRecord", &BaseTable::makeRecord);
+    cls.def("copyRecord",
+            (std::shared_ptr<BaseRecord> (BaseTable::*)(BaseRecord const &)) &BaseTable::copyRecord);
+    cls.def("copyRecord",
+            (std::shared_ptr<BaseRecord> (BaseTable::*)(BaseRecord const &, SchemaMapper const &))
+                &BaseTable::copyRecord);
+    cls.def("getSchema", &BaseTable::getSchema);
+    cls.def_property_readonly("schema", &BaseTable::getSchema);
+    cls.def("getBufferSize", &BaseTable::getBufferSize);
+    cls.def("clone", &BaseTable::clone);
+    cls.def("preallocate", &BaseTable::preallocate);
+
+    return cls;
+}
+
+PYBIND11_PLUGIN(base) {
+    py::module mod("base");
+
+    py::module::import("lsst.afw.table.schema");
+    py::module::import("lsst.afw.table.baseColumnView");
```

Comment:
@r-owen was saying he liked it better this way, and I'm inclined to agree.  I don't think it's a big enough deal to put a lot of effort into standardizing, though.